core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt

```diff
@@ -0,0 +1,26 @@
+package app.k9mail.core.ui.compose.theme.color
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.luminance
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
+
+private const val MAX_DARK_COLOR_LUMINANCE = 0.5f
+
+@Composable
+fun SetSystemBarsColor(
+    color: Color,
+) {
+    val systemUiController = rememberSystemUiController()
+    val isDarkColor = color.luminance() < MAX_DARK_COLOR_LUMINANCE
+
+    DisposableEffect(Unit) {
+        systemUiController.setSystemBarsColor(
+            color = color,
+            darkIcons = !isDarkColor,
```

Comment:
It took me a moment to understand this. Maybe `isDarkColor` should be `isDarkBackgroundColor`?

Reply:
But the `SystemUiController` uses color as terminology and we could cut the discussion here as it applies the luminance check by default anyway. So this check could go.